core/src/main/java/com/sequenceiq/cloudbreak/authorization/ReturnValuePermissionChecker.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.cloudbreak.authorization;
+
+import java.lang.annotation.Annotation;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.aspect.organization.CheckPermissionsByReturnValue;
+import com.sequenceiq.cloudbreak.domain.security.User;
+import com.sequenceiq.cloudbreak.validation.OrganizationPermissions.Action;
+import com.sequenceiq.cloudbreak.validation.OrganizationPermissions.Resource;
+
+@Component
+public class ReturnValuePermissionChecker implements PermissionChecker<CheckPermissionsByReturnValue> {
+
+    @Inject
+    private PermissionCheckingUtils permissionCheckingUtils;
+
+    @Override
+    public <T extends Annotation> Object checkPermissions(T rawMethodAnnotation, Resource resource, User user,
+            ProceedingJoinPoint proceedingJoinPoint, MethodSignature methodSignature) {
+
+        CheckPermissionsByReturnValue methodAnnotation = (CheckPermissionsByReturnValue) rawMethodAnnotation;
+        Action action = methodAnnotation.action();
+        Object proceed = permissionCheckingUtils.proceed(proceedingJoinPoint, methodSignature);
+        if (proceed instanceof Optional<?>) {
+            Optional<?> optionalResult = (Optional<?>) proceed;
+            if (optionalResult.isPresent()) {
```

Comment:
I flipped the if statement and the result is the following:
```
        if (proceed instanceof Optional<?>) {
            Optional<?> optionalResult = (Optional<?>) proceed;
            if (!optionalResult.isPresent()) {
                return proceed;
            }
            permissionCheckingUtils.checkPermissionsByTarget(optionalResult.get(), user, resource, action);
        }
        permissionCheckingUtils.checkPermissionsByTarget(proceed, user, resource, action);
```
It seems there are cases where `permissionCheckingUtils.checkPermissionsByTarget()` happens twice